### PR TITLE
Port 'Execute SQL' algorithm and widget wrapper to c++ 

### PR DESCRIPTION
**Pull request by nyalldawson**
## Description

The c++ algorithm is close to a direct port of the Python one, with the addition of improved help strings and per-parameter help, and some extra feedback (showing the actual SQL to be run).

The widget wrapper is considerably simplified from the Python version. The python version was needlessly complex, as a result of API introduced in the original PR which duplicated existing generic functionality from the modeler API. I've removed that, and deprecated the duplicate API.

There's also new tests for the widget wrapper (none were present before), and a new test for the algorithm itself.
 
## AI tool usage

 - [x] AI tool(s): Gemini was used to do the initial algorithm port + write the new help strings, revised by hand by myself.

### CODE REVIEW

File: src/analysis/processing/qgsalgorithmexecutesql.cpp [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+    if ( !vectorLayer || !vectorLayer->isValid() )
+    {
+      continue;
+    }
+
+    // Issue https://github.com/qgis/QGIS/issues/24041
+    // When using this algorithm from the graphic modeler, it may try to
+    // access (thanks the QgsVirtualLayerProvider) to memory layer that
+    // belongs to temporary QgsMapLayerStore, not project.
+    // So, we write them to disk is this is the case.
```
Comment (alexbruy) [inline src/analysis/processing/qgsalgorithmexecutesql.cpp]:
```suggestion
    // So, we write them to disk if this is the case.
```

**Review by alexbruy — APPROVED** [review]
